src/box/lua/error.cc

```diff
@@ -382,6 +382,21 @@ luaT_error_set(struct lua_State *L)
 	return 0;
 }
 
+/**
+ * Return whether first argument is box error. Raise error if invoked
+ * with zero or more then one arguments.
+ */
+static int
+luaT_error_is(struct lua_State *L)
+{
+	if (lua_gettop(L) != 1) {
+		diag_set(IllegalParams, "Usage: box.error.is(error)");
+		luaT_error(L);
+	}
```

Comment:
`box.tuple.is()` works without arguments and ignores extra arguments. I think `box.error.is()` should behave in the same way.

Reply:
Fixed. The commit is moved to a distinct PR.

<details>
<summary>diff</summary>

```diff
diff --git a/src/box/lua/error.cc b/src/box/lua/error.cc
index 2eb487d96..bfa077de1 100644
--- a/src/box/lua/error.cc
+++ b/src/box/lua/error.cc
@@ -389,11 +389,7 @@ luaT_error_set(struct lua_State *L)
 static int
 luaT_error_is(struct lua_State *L)
 {
-       if (lua_gettop(L) != 1) {
-               diag_set(IllegalParams, "Usage: box.error.is(error)");
-               luaT_error(L);
-       }
-       lua_pushboolean(L, luaL_iserror(L, 1) != NULL);
+       lua_pushboolean(L, lua_gettop(L) >= 1 && luaL_iserror(L, 1) != NULL);
        return 1;
 }

diff --git a/test/box-luatest/error_subsystem_improvements_test.lua b/test/box-luatest/error_subsystem_improvements_test.lua
index 1140ea56c..3ba7bce7c 100644
--- a/test/box-luatest/error_subsystem_improvements_test.lua
+++ b/test/box-luatest/error_subsystem_improvements_test.lua
@@ -1035,13 +1035,12 @@ g.after_test('test_dml_upsert_error', function(cg)
 end)

 g.test_error_is = function()
-    t.assert_error_msg_equals('Usage: box.error.is(error)', box.error.is)
-    t.assert_error_msg_equals('Usage: box.error.is(error)', box.error.is, 1, 2)
-
     t.assert(box.error.is(box.error.new(box.error.UNKNOWN)))
+    t.assert(box.error.is(box.error.new(box.error.UNKNOWN)), 'foo')

     t.assert_not(box.error.is(nil))
     t.assert_not(box.error.is(1))
-    t.assert_not(box.error.is('bar'))
+    t.assert_not(box.error.is('foo'))
+    t.assert_not(box.error.is('foo'), 'bar')
     t.assert_not(box.error.is(box.tuple.new({1, 'foo'})))
```

</details>